plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -191,9 +192,49 @@ public Optional<RemoteDatabaseObject> toRemoteTable(ConnectorSession session, St
         return toRemoteTable(projectId, remoteDatasetName, tableName, () -> findTableIdsIgnoreCase(session, DatasetId.of(projectId, remoteDatasetName), tableName));
     }
 
-    public Optional<RemoteDatabaseObject> toRemoteTable(String projectId, String remoteDatasetName, String tableName, Iterable<TableId> tableIds)
+    /**
+     * Similar to BigQueryClient#toRemoteTable method, but this iterates through all tableIds filtering out
+     * the ambiguous tables when caseInsensitiveNameMatching is enabled, additionally mapping them to a SchemaTableName
+     */
+    public List<SchemaTableName> getNonAmbiguousSchemaTableNames(String projectId, String remoteDatasetName, Iterable<TableId> tableIds)
```

Comment:
https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-get-in-method-names-unless-an-object-must-be-a-java-bean
```suggestion
    public List<SchemaTableName> listNonAmbiguousSchemaTableNames(String projectId, String remoteDatasetName, Iterable<TableId> tableIds)
```